lms/djangoapps/sneakpeek_deeplink/middleware.py

```diff
@@ -60,6 +65,12 @@ def process_request(self, request):
         if not CoursePreference.course_allows_nonregistered_access(course_id):
             return None
 
+        if not modulestore().has_course(course_id):
+            msg = u"""Sneakpeek attempt for non-existent 
+                course %s with lingering preferences""", course_id
```

Comment:
1. This is the old-style, deprecated form of Python string interpolation; don't use it.
If you're formatting strings, always use the form:
```python
message = "foo {course_id} bar".format(course_id=course_id)
```

2. Don't use docstrings for variable assignment. If you need multiline definitions,
use Python's implicit string concatenation:
```python
message = (
    "Line one "
    "and "
    "line two"
)
```

3. While this example isn't too bad, don't abbreviate variable names: use `message` instead of `msg`.
Always favor clarity over brevity.

4. Logging calls can (and should) actually be lazily interpolated (yes, using the old format):

```python
message = u"Sneakpeek attempt for non-existent course %s"
log.warning(message, course_id)
```